### ISSUE TITLE
3 tests are failing/erroring

### ISSUE THREAD

**Issue by hroncok** [necaris/python3-openid#4]
Hi, when I run `python3 -m unittest openid.test.test_suite`, I got the following Traceback:

````
Traceback (most recent call last):
  File "/usr/lib64/python3.3/runpy.py", line 160, in _run_module_as_main
    "__main__", fname, loader, pkg_name)
  File "/usr/lib64/python3.3/runpy.py", line 73, in _run_code
    exec(code, run_globals)
  File "/usr/lib64/python3.3/unittest/__main__.py", line 19, in <module>
    main(module=None)
  File "/usr/lib64/python3.3/unittest/main.py", line 124, in __init__
    self.parseArgs(argv)
  File "/usr/lib64/python3.3/unittest/main.py", line 168, in parseArgs
    self.createTests()
  File "/usr/lib64/python3.3/unittest/main.py", line 175, in createTests
    self.module)
  File "/usr/lib64/python3.3/unittest/loader.py", line 137, in loadTestsFromNames
    suites = [self.loadTestsFromName(name, module) for name in names]
  File "/usr/lib64/python3.3/unittest/loader.py", line 137, in <listcomp>
    suites = [self.loadTestsFromName(name, module) for name in names]
  File "/usr/lib64/python3.3/unittest/loader.py", line 122, in loadTestsFromName
    test = obj()
  File "./openid/test/__init__.py", line 182, in test_suite
    combined_suite.addTests(pyUnitTests())
  File "./openid/test/__init__.py", line 117, in pyUnitTests
    suite.addTest(mod.pyUnitTests())
  File "./openid/test/trustroot.py", line 76, in pyUnitTests
    test_data = test_data_file.read()
  File "/usr/lib64/python3.3/encodings/ascii.py", line 26, in decode
    return codecs.ascii_decode(input, self.errors)[0]
UnicodeDecodeError: 'ascii' codec can't decode byte 0xcf in position 460: ordinal not in range(128)
````

The character causing that is pi in test/data/trustroot.txt.

If I sed that line away, the tests are failing anyway:

````
python3 -m unittest openid.test.test_suite
adding openid to sys.path
./openid/test/__init__.py:83: UserWarning: Could not import twill; skipping test_examples.
  warnings.warn("Could not import twill; skipping test_examples.")
.................../openid/server/server.py:980: DeprecationWarning: The "namespace" attribute of CheckIDRequest objects is deprecated. Use "message.getOpenIDNamespace()" instead
  self.fields = Message(request.namespace)
.................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................../openid/test/test_pape_draft5.py:306: UserWarning: "none" used as a policy URI (see PAPE draft < 5)
  self.resp.parseExtensionArgs(args, is_openid1=False)
............................................................................................................................................................/openid/test/storetest.py:258: UserWarning: Could not import MySQLdb. Skipping MySQL store tests.
  warnings.warn("Could not import MySQLdb. Skipping MySQL store tests.")
../openid/test/storetest.py:326: UserWarning: Could not import psycopg. Skipping PostgreSQL store tests.
  warnings.warn("Could not import psycopg. Skipping PostgreSQL store tests.")
..................................................................................../openid/test/test_fetchers.py:112: UserWarning: Skipping tests for 'pycurl' fetcher because the library did not import.
  'the library did not import.' % (library_name,))
./openid/test/test_fetchers.py:90: ResourceWarning: unclosed <socket.socket object, fd=4, family=2, type=1, proto=6>
  assert should_raise_exc
F..............................................................................................................................................................................................................................................................................................................................................................................................E
======================================================================
ERROR: unittest.case.FunctionTestCase (djangoExampleTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "./openid/test/__init__.py", line 155, in djangoExampleTests
    import django.test.simple
  File "/usr/lib/python3.3/site-packages/django/test/__init__.py", line 5, in <module>
    from django.test.client import Client, RequestFactory
  File "/usr/lib/python3.3/site-packages/django/test/client.py", line 16, in <module>
    from django.contrib.auth import authenticate, login
  File "/usr/lib/python3.3/site-packages/django/contrib/auth/__init__.py", line 5, in <module>
    from django.middleware.csrf import rotate_token
  File "/usr/lib/python3.3/site-packages/django/middleware/csrf.py", line 16, in <module>
    from django.utils.cache import patch_vary_headers
  File "/usr/lib/python3.3/site-packages/django/utils/cache.py", line 26, in <module>
    from django.core.cache import get_cache
  File "/usr/lib/python3.3/site-packages/django/core/cache/__init__.py", line 70, in <module>
    if DEFAULT_CACHE_ALIAS not in settings.CACHES:
  File "/usr/lib/python3.3/site-packages/django/conf/__init__.py", line 53, in __getattr__
    self._setup(name)
  File "/usr/lib/python3.3/site-packages/django/conf/__init__.py", line 48, in _setup
    self._wrapped = Settings(settings_module)
  File "/usr/lib/python3.3/site-packages/django/conf/__init__.py", line 132, in __init__
    mod = importlib.import_module(self.SETTINGS_MODULE)
  File "/usr/lib/python3.3/site-packages/django/utils/importlib.py", line 35, in import_module
    __import__(name)
  File "/home/churchyard/rpmbuild/BUILD/python3-openid-3.0.3/examples/djopenid/settings.py", line 9
    except ImportError, e:
                      ^
SyntaxError: invalid syntax

======================================================================
FAIL: unittest.case.FunctionTestCase (test)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "./openid/test/test_fetchers.py", line 14, in _assertEqual
    assert v1 == v2
AssertionError

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "./openid/test/test_fetchers.py", line 228, in test
    run_fetcher_tests(server)
  File "./openid/test/test_fetchers.py", line 125, in run_fetcher_tests
    test_fetcher(f, True, server)
  File "./openid/test/test_fetchers.py", line 70, in test_fetcher
    failUnlessResponseExpected(expected, actual, extra=locals())
  File "./openid/test/test_fetchers.py", line 22, in failUnlessResponseExpected
    _assertEqual(expected.body, actual.body, extra)
  File "./openid/test/test_fetchers.py", line 16, in _assertEqual
    raise AssertionError("%r != %r ; context %r" % (v1, v2, extra))
AssertionError: '/success' != b'/success' ; context {'plain': <function test_fetcher.<locals>.plain at 0x7f0ccc41e830>, 'actual': <HTTPResponse status 200 for http://127.0.0.1:5058/success>, 'fetcher': <openid.fetchers.HTTPLib2Fetcher object at 0x7f0ccc42eb50>, 'path': '/success', 'cases': [('/success', <HTTPResponse status 200 for http://127.0.0.1:5058/success>), ('/301redirect', <HTTPResponse status 200 for http://127.0.0.1:5058/success>), ('/302redirect', <HTTPResponse status 200 for http://127.0.0.1:5058/success>), ('/303redirect', <HTTPResponse status 200 for http://127.0.0.1:5058/success>), ('/307redirect', <HTTPResponse status 200 for http://127.0.0.1:5058/success>), ('/notfound', <HTTPResponse status 404 for http://127.0.0.1:5058/notfound>), ('/badreq', <HTTPResponse status 400 for http://127.0.0.1:5058/badreq>), ('/forbidden', <HTTPResponse status 403 for http://127.0.0.1:5058/forbidden>), ('/error', <HTTPResponse status 500 for http://127.0.0.1:5058/error>), ('/server_error', <HTTPResponse status 503 for http://127.0.0.1:5058/server_error>)], 'geturl': <function test_fetcher.<locals>.geturl at 0x7f0ccc41e8c0>, 'fetch_url': 'http://127.0.0.1:5058/success', 'should_raise_exc': True, 'expected_headers': {'content-type': 'text/plain'}, 'server': <http.server.HTTPServer object at 0x7f0ccc42ec10>, 'expect_success': <HTTPResponse status 200 for http://127.0.0.1:5058/success>, 'expected': <HTTPResponse status 200 for http://127.0.0.1:5058/success>}

----------------------------------------------------------------------
Ran 1331 tests in 1.328s

FAILED (failures=1, errors=1)
````

Am I doing anything wrong? As SyntaxError is something that should not happen during the tests, right?

**Comment by moreati** [comment]
This should be fixed soon -  `openid.fetchers.CurlHTTPFetcher` was fixed by #7 and `openid.fetchers.HTTPLib2Fetcher` will be fixed when #11 is merged

**Comment by hroncok** [comment]
:+1:

**Comment by necaris** [comment]
Closed by #11
